consensus/e2e_tests/utils_test.go

```diff
@@ -526,40 +536,328 @@ func baseRpcMock(t *testing.T, _ modules.EventsChannel) *mockModules.MockRPCModu
-			maxHeight := consensusMod.GetAggregatedStateSyncMetadataMaxHeight()
-			// TECHDEBT(#352): The asynchronicity of this leads to a non-deterministic failing test.
-			// See this discussion for details: https://github.com/pokt-network/pocket/pull/528/files#r1150711575
-			consensusMod.SetHeight(maxHeight)
-			return nil
-		case coreTypes.StateMachineEvent_Consensus_IsSyncedValidator:
-			t.Logf("Mocked validator node is synced")
-			return nil
-		case coreTypes.StateMachineEvent_Consensus_IsSyncedNonValidator:
-			t.Logf("Mocked non-validator node is synced")
-			return nil
-		default:
-			log.Printf("Mocked node is not handling this event: %s", event)
-			return nil
+func waitForNextBlock(t *testing.T,
+	clck *clock.Mock,
+	eventsChannel modules.EventsChannel,
+	pocketNodes IdToNodeMapping,
+	height uint64,
+	step uint8,
+	round uint8,
+	noOfExpectedMsgs int,
+	maxWaitTime time.Duration,
+	failOnExtraMessages bool) []*anypb.Any {
+
+	leaderId := typesCons.NodeId(pocketNodes[1].GetBus().GetConsensusModule().GetLeaderForView(height, uint64(round), step))
+	newRoundMessages := waitForNewRound(t, clck, eventsChannel, pocketNodes, height, step, round, noOfExpectedMsgs, maxWaitTime, failOnExtraMessages)
+	prepareProposal := waitForPrepareProposal(t, clck, eventsChannel, pocketNodes, newRoundMessages, height, step+1, round, leaderId, noOfExpectedMsgs, maxWaitTime, failOnExtraMessages)
+	prepareVote := waitForPrepareVotes(t, clck, eventsChannel, pocketNodes, prepareProposal, noOfExpectedMsgs, maxWaitTime, failOnExtraMessages)
+	preCommitProposal := waitForPreCommit(t, clck, eventsChannel, pocketNodes, prepareVote, height, step+2, round, leaderId, noOfExpectedMsgs, maxWaitTime, failOnExtraMessages)
+	commitProposal := waitForCommit(t, clck, eventsChannel, pocketNodes, preCommitProposal, height, step+3, round, leaderId, noOfExpectedMsgs, maxWaitTime, failOnExtraMessages)
+	return waitForDecide(t, clck, eventsChannel, pocketNodes, commitProposal, height, step+4, round, leaderId, noOfExpectedMsgs, maxWaitTime, failOnExtraMessages)
+}
+
+func waitForNewRound(t *testing.T,
+	clck *clock.Mock,
+	eventsChannel modules.EventsChannel,
+	pocketNodes IdToNodeMapping,
+	height uint64,
+	step uint8,
+	round uint8,
+	noOfExpectedMsgs int,
+	maxWaitTime time.Duration,
+	failOnExtraMessages bool) []*anypb.Any {
+
+	newRoundMessages, err := WaitForNetworkConsensusEvents(t, clck, eventsChannel, consensus.NewRound, consensus.Propose, noOfExpectedMsgs*noOfExpectedMsgs, maxWaitTime, failOnExtraMessages)
+	require.NoError(t, err)
+	for nodeId, pocketNode := range pocketNodes {
+		nodeState := GetConsensusNodeState(pocketNode)
+		assertNodeConsensusView(t, nodeId,
+			typesCons.ConsensusNodeState{
+				Height: height,
+				Step:   step,
+				Round:  round,
+			},
+			nodeState)
+		require.Equal(t, false, nodeState.IsLeader)
+		require.Equal(t, typesCons.NodeId(0), nodeState.LeaderId)
+	}
+
+	return newRoundMessages
+}
+
+func waitForPrepareProposal(t *testing.T,
```

Comment:
This is a good approach, but you're making it too specific. 

Instead of `WaitForXXXXProposal`  or `waitForXXXVote`, just implement `WaitForProposalMsgs` and `WaitForVoteMsgs` and make the `Step` a parameter. It'll reduce the code by a lot and make it easier to read.